### DISCUSSION TITLE
Instant search based on search template FSE

### DISCUSSION THREAD

**Discussion by r-chrzan** [WordPress/gutenberg#56751] [Interactivity API]
Is it possible at the current stage of work to achieve the effect of a instant search engine based on a search template in editor, as is the case with wpmovies?

**Comment by luisherranz** [comment]
We haven't tried it yet, but we think it should be possible to build it for the Query Loop, in a similar way [to the client-side pagination](https://github.com/WordPress/gutenberg/pull/53812), except that the `navigate` calls are triggered from the search block instead of the pagination links.

**Reply by r-chrzan** [reply]
@luisherranz I'm working on one project as part of Interactivity tests and I managed to achieve the effect of searching in the Query Loop block, but it seems to work a little slower due to the (promises?), and I can't get full-page navigation in any way because there is no such thing at the moment options? Am I right?



https://github.com/WordPress/gutenberg/assets/26217683/f7a82dd9-13c1-4995-a0db-22642dbf9a25

**Reply by luisherranz** [reply]
@r-chrzan That's right, _full-page_ client-side navigation is not available yet in Gutenberg. It will likely take a bit longer as it's something very complex to release and we need to do it carefully. That said, I think you should be able to achieve what you want with the _region-based_ client-side navigation.

Can you show the code that you are using so I can take a look?

**Reply by r-chrzan** [reply]
@luisherranz If you want to help with this particular case I can send you a whole site exported from LocalWP, because for now i only check possibilities of using Interactivity, or you can provide me how to achieve display post when i have prefetching and client side nav. I based on code from docs and a bit of wpmovies

**Comment by r-chrzan** [comment]
Hello to everyone who is following this thread. Finally, I managed to achieve the effect of a instant search region-based(Query loop block) engine using the interactivity API(after public release v 1.0), using the pre_render_block and query_loop_block_query_vars filters. I can share an example plugin somewhere if anyone is interested. I am sharing a short video

https://github.com/WordPress/gutenberg/assets/26217683/269aa824-1055-45e8-9b3c-8dcb1c1136b5


cc: @luisherranz

**Reply by cbravobernal** [reply]
Of course we are interested!

Feel free to share in the showcase discussion: https://github.com/WordPress/gutenberg/discussions/55642

**Reply by luisherranz** [reply]
That's pretty cool to see 😄👏

**Reply by r-chrzan** [reply]
Posted 💪🏻 [here](https://github.com/WordPress/gutenberg/discussions/55642#discussioncomment-9667164)

**Reply by michalczaplinski** [reply]
Really nice stuff @r-chrzan! 👏 

Just a very minor nitpick:

You don't need to import the router inside of `*updateSearch()` because it's not used there. You can instead put the import statement inside of `updateURL()` and use it there:

```js
// We can use `await` here instead of `yield` because it's inside of a
// regular async function, not inside of a store action.
const { actions } = await import('@wordpress/interactivity-router');
actions.navigate(`${window.location.pathname}${url.search}`);
```

**Reply by r-chrzan** [reply]
@michalczaplinski thanks for a review! I pushed that small changes!
